threading/client/client.py

```diff
@@ -0,0 +1,396 @@
+                err_code = -1
+                size = struct.pack("i", err_code)
+                sent = cli_socket.send(size)
+        else:
+            print(f"{dir} is not a valid directory location...")
+            err_code = -1
+            size = struct.pack("i", err_code)
+            sent = cli_socket.send(size)
+
+            if sent == 0:
+                raise RuntimeError("Socket connection broken: invalid upload file")
+    else:
+        raise RuntimeError("Invalid menu option received: upload file")
+
+def recv_file_list():
+    '''
+    RECV_FILE_LIST:
+        PARAMETER: None
+        BRIEF: receives file list from the server and prints them to the user
+        RETURN: None
+    '''
+    file_count = cli_socket.recv(4)
+    file_count = int.from_bytes(file_count, "little", signed=True)
+    received = 0
+    data = b''
+
+    while received < file_count:
+        file_len = cli_socket.recv(8)
+        file_len = int.from_bytes(file_len, "little", signed=True)
+        data = cli_socket.recv(file_len)
+        file_list.append(data.decode('utf-8'))
+        received += 1
+    print(file_list)
+
+def get_file_list(menu_option):
+    '''
+    GET_FILE_LIST:
+        PARAMETER: MENU_OPTION - menu option 3 for list file server directory
+        BRIEF: sends command to the server to obtain list of current files
+        RETURN: None
+    '''
+    if menu_option == 3:
+        request = 100
+        list_dir = struct.pack("I", request)
+        sent = cli_socket.send(list_dir)
+
+        if sent == 0:
+            raise RuntimeError("Socket connection broken: get dir list")
+
+        print("Sent list directory command to server...")
+        totalsent = 0
+        list_cmd = "ls"
+
+        while totalsent < len(list_cmd):
+            sent = cli_socket.send(list_cmd[totalsent:].encode('utf-8'))
+            if sent == 0:
+                raise RuntimeError("Socket connection broken: sending list command")
+            totalsent += sent
+        recv_file_list()
+    else:
+        raise RuntimeError("Invalid menu option received: get file")
+    file_list.clear()
+
+if __name__ == "__main__":
+    '''
+    MAIN:
+        BRIEF: main method to call client application functionality
+        USAGE: calls all necessary methods and disallows invalid input from the user
+    '''
+    signal.signal(signal.SIGINT, signal_handler)
+    while True:
+        try:
+            user_in = main_menu()
+            if user_in == 1:
+                download_existing_file(user_in)
+            elif user_in == 2:
+                upload_file(user_in)
+            elif user_in == 3:
+                get_file_list(user_in)
+            elif user_in == 4:
+                help_option(user_in)
+            elif user_in == 5:
+                exit_server(user_in)
+                exit()
+            elif user_in > 5 or user_in <= 0:
+                print("Please select an appropriate menu option...")
+        except TypeError:
```

Comment:
If the user types in something besides a number, and it just exits, that seems a bit abrupt / not user-friendly. Maybe a usage or help message?